module-2/lab-pivot-table-and-correlation/your_code/main.ipynb

```diff
@@ -21,13 +21,14 @@
   },
```

Comment:
- It doesn't make sense to fill missing std with a 0 as commented today in class. Hope the explanation clarifies the concept. 
- What can you say about Minute Very Active and Steps? Write a comment below
"Existe una correlación lineal positiva, no excesivamente fuerte ya que es 0,50. Según la gráfica, una persona puede estar activa y no necesariamente dar pasos" . El gráfico muestra una cierta relación positiva entre número de minutos y pasos, a más minutos activo más pasos. Sin embargo, el gráfico no muestra una relación "muy fuerte" como demuestra un coef de corr de 0.5. 
- Good one!! "No existe ninguna relación entre los pasos y los minutos de sedentarismo. La correlación es muy cercana al 0 (0.076) y como vemos en el gráfico ambas variables no se relacionan."
- You haven't commented this one Use matplotlib to visually plot MinutesOfSleep vs Steps. Based on the results of the computed correlation and the plot. What can you say about these 2 variables? however, it's pretty close to the relationship between minutes sedentary and steps. 
- Based on the result of the correlation matrix. What are the highly correlated features with the variable Steps?: you only need to sort the Steps column of your correlation matrix not the entire matrix.
- Use matplotlib to visually plot Study time vs Grade. Based on the results of the computed correlation and the plot. What can you say about these 2 variables? Your comment "Existe una correlación lineal positiva muy leve entre el tiempo de estudio y la puntación de los estudiantes. Ya que entre 8 y 12,5 horas de estudio aproximadamente, no hay una relación postiva, se mantiene la misma puntuación aunque aumentes las horas entre 8 y 12,5" is not quite correct. Existe una relación positiva entre la nota y las horas de estudio, a más horas de estudio más nota, no obstante este patrón parece desvanecerse a partir de las 8/9 horas de estudio. A partir de este punto las notas no se incrementan con más horas de estudio. Así lo demuestra las dos corr, Spearman y Pearson. 
**Note: Spearman vs Pearson: It is usually a little bit higher Spearman than Pearson but it doesn't mean necessarily that the relationship is not linear. Why? The Pearson is most appropriate for measurements taken from an interval scale (temperature, dates, lengths, etc), while the Spearman is best for measurements taken from ordinal scales (rank orders, spectrum of values (agree, neutral, disagree), or healthy vs non-healthy). However, with large samples, the correlation coefficients will be similar. The largest differences will be seen with small sample sizes. Pearson's assumes constant variance and linearity. If your variable violates that, Spearman's would be the best to use. This is the best explanation I have seen up to now (extract from quora). In other words, Spearman's coefficient measures the rank order of the points (it doesn't care where the points are) while Pearson measures the linear relationship between the two. In the end, it's not about a better or worse correlation, it's more about that we should use one or another depending on the characteristics of our variables.
In statistics and generally speaking, the fact two variables correlate cannot prove anything - only further research can actually prove that one thing affects the other.Data reliability is related to the size of the sample. The more data you collect, the more reliable your result.**